src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -148,6 +148,25 @@ private async Task CreateBlocksAsync()
                         continue;
                     }
 
+                    // Check if the current version of the node is reaching end of life.
+                    ChainedHeader tip = this.consensusManager.Tip;
+                    int maxSupportedBlockHeightGracePeriod = this.network.Consensus.Options.MaxSupportedBlockHeightGracePeriod;
+                    int maxSupportedMineBlockHeight = this.network.ConsensusOptions.MaxSupportedMineBlockHeight;
+                    if ((maxSupportedBlockHeightGracePeriod > 0) && (tip.Height > (maxSupportedMineBlockHeight - maxSupportedBlockHeightGracePeriod)))
+                    {
+                        this.logger.LogWarning("The current version of the node is reaching its end of life. Please upgrade the node to the latest version in order to retain ability to mine PoA blocks.");
+                    }
+
+                    // Don't mine blocks if our client version is too old.
+                    if (maxSupportedMineBlockHeight > tip.Height)
```

Comment:
Isn't that supposed to be reversed and checked for 0: `if (maxSupportedMineBlockHeight > 0 && maxSupportedMineBlockHeight < tip.Height)`?